codegen/gen.py

```diff
@@ -1015,6 +1025,7 @@ def main() -> None:
             kernel_index=kernel_index,
             manual_registration=options.manual_registration,
             add_exception_boundary=options.add_exception_boundary,
+            lib_name=options.lib_name,
```

Comment:
Let's error out, if `lib_name` is specified but `manual_registration` is not enabled.